docs/semantics.md

```diff
@@ -95,7 +95,16 @@ The negative result for ```foo/``` will be cached, but that only helps with the
 To alleviate this, Cloud Storage FUSE supports a "type cache" on directory inodes. When type cache is enabled, each directory inode will maintain a mapping from the name of its children to whether those children are known to be files or directories or both. When a child is looked up, if the parent's cache says that the child is a file but not a directory, only one Cloud Storage object will need to be stated. Similarly if the child is a directory but not a file.
```

Comment:
can we include "metadata-cache:type-cache-max-size-mb" in existing e2e tests? IMO, we should include in the following packages - operations, implicit-dir, explicit-dir, local-file, readonly.

Should be added similar to https://github.com/GoogleCloudPlatform/gcsfuse/blob/read_cache_release/tools/integration_tests/operations/operations_test.go#L105